### PR TITLE
Add support for parallel exports

### PR DESCRIPTION
**Pull request by anish749**
First draft for adding support for parallel exports based on an int / long splitting column.

Implements #51

### CODE REVIEW

**Comment by anish749** [comment]
PTAL @labianchin

File: dbeam-core/src/main/java/com/spotify/dbeam/args/QueryBuilderArgs.java [hunk 1]
```diff
@@ -78,29 +96,101 @@ private static Boolean checkTableName(String tableName) {
...
-              return String.format(" WHERE %s >= '%s' AND %s < '%s'",
-                                   partitionColumn, datePartition, partitionColumn, nextPartition);
+              return String.format(" AND %s >= '%s' AND %s < '%s'",
+                  partitionColumn, datePartition, partitionColumn, nextPartition);
             })
     ).orElse("");
-    return Lists.newArrayList(
-        String.format("SELECT * FROM %s%s%s", this.tableName(), where, limit));
+
+    if (parallelism().isPresent() && splitColumn().isPresent()) {
```
Comment (labianchin) [inline dbeam-core/src/main/java/com/spotify/dbeam/args/QueryBuilderArgs.java]: Maybe extract this `if` branch into separate method?

File: dbeam-core/src/main/java/com/spotify/dbeam/args/QueryBuilderArgs.java [hunk 2]
```diff
@@ -78,29 +97,159 @@ private static Boolean checkTableName(String tableName) {
...
+        splitColumn,
+        tableName,
+        partitionCondition);
+    long min;
+    long max;
+    try (Statement statement = connection.createStatement()) {
+      final ResultSet
+          resultSet =
+          statement.executeQuery(query);
+      resultSet.first();
```
Comment (labianchin) [inline dbeam-core/src/main/java/com/spotify/dbeam/args/QueryBuilderArgs.java]:
```
[main] INFO com.spotify.dbeam.avro.BeamJdbcAvroSchema - Elapsed time to schema 0.452 seconds
[main] ERROR com.spotify.dbeam.jobs.ExceptionHandling - Failure:
org.postgresql.util.PSQLException: Operation requires a scrollable ResultSet, but this ResultSet is FORWARD_ONLY.
        at org.postgresql.jdbc.PgResultSet.checkScrollable(PgResultSet.java:280)
        at org.postgresql.jdbc.PgResultSet.first(PgResultSet.java:355)
        at com.spotify.dbeam.args.QueryBuilderArgs.findSplitLimits(QueryBuilderArgs.java:178)
        at com.spotify.dbeam.args.QueryBuilderArgs.buildQueries(QueryBuilderArgs.java:139)
        at com.spotify.dbeam.jobs.JdbcAvroJob.prepareExport(JdbcAvroJob.java:93)
        at com.spotify.dbeam.jobs.JdbcAvroJob.runExport(JdbcAvroJob.java:134)
        at com.spotify.dbeam.jobs.JdbcAvroJob.main(JdbcAvroJob.java:142)
```

Maybe use `checkState(resultSet.next(), "Min/Max query returned empty results");` instead.

Comment (anish749) [reply dbeam-core/src/main/java/com/spotify/dbeam/args/QueryBuilderArgs.java]: Good catch.. I'll add a check

**Comment by anish749** [comment]
@labianchin this is now ready for merging. I would update the documentation and add best practices separately.

**Comment by labianchin** [comment]
:shipit: 

Let's ship this, try a bit and then iterate a bit more on it. We might still be missing metrics and maybe some tests. But given this is a new feature and it does not break existing flows, we can ship and try.

**Pull request merged**